File: app/src/main/java/com/ddiehl/android/reversi/view/SettingsActivity.java
package com.ddiehl.android.reversi.view;

import android.app.Activity;
import android.os.Bundle;

public class SettingsActivity extends Activity {
    private static final String TAG = SettingsActivity.class.getSimpleName();

    public static final String EXTRA_SETTINGS_MODE = "settings_mode";
    public static final String EXTRA_IS_SIGNED_IN = "is_signed_in";
    public static final String EXTRA_SIGNED_IN_ACCOUNT = "signed_in_account";

    public static final String PREF_PLAY_SERVICES_SIGN_IN = "pref_play_services_sign_in";
    public static final String PREF_PLAYER_NAME = "pref_player_name";
    public static final String PREF_AI_DIFFICULTY = "pref_ai_difficulty";

    public static final int SETTINGS_MODE_SINGLE_PLAYER = 101;
    public static final int SETTINGS_MODE_MULTI_PLAYER = 102;
    public static final int RESULT_SIGN_IN = 201;
    public static final int RESULT_SIGN_OUT = 202;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getFragmentManager().beginTransaction()
                .replace(android.R.id.content, SettingsFragment.newInstance(getIntent())).commit();
    }
}

File: app/src/main/java/com/ddiehl/android/reversi/settings/SettingsActivity.kt
package com.ddiehl.android.reversi.settings

import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import butterknife.bindView
import com.ddiehl.android.reversi.R

class SettingsActivity : AppCompatActivity() {

    companion object {
        val SETTINGS_MODE_SINGLE_PLAYER = 101
        val SETTINGS_MODE_MULTI_PLAYER = 102
        val RESULT_SIGN_OUT = 202
    }

    val mToolbar by bindView<Toolbar>(R.id.toolbar)
    val mSettingsView by bindView<SettingsView>(R.id.settings_view)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_activity)
        window.decorView.setBackgroundColor(ContextCompat.getColor(this, R.color.gray90))

        setSupportActionBar(mToolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        setTitle(R.string.settings)
    }

    override fun onStart() {
        super.onStart()
        bindUi()
    }

    private fun bindUi() {
        val singlePlayerSettings = SinglePlayerSettings(this)
        mSettingsView.bindSettings(singlePlayerSettings)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                if (!isFinishing) finish()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }
}
